### PR TITLE
#570 trigonometry and math functions

### PR DESCRIPTION
**Pull request by imaqtkatt**

### CODE REVIEW

File: src/fun/builtins.bend [hunk 1]
```diff
@@ -383,3 +383,60 @@ Bytes/decode_ascii (List/Nil)       = (String/Nil)
...
+Math/PI = 3.1415926535
+
+# Math/sin(a: f24) -> f24
+# Computes the sine of the given angle in radians.
+hvm Math/sin:
+  ($([<<0x0] a) a)
+
+# Math/cos(a: f24) -> f24
+# Computes the cosine of the given angle in radians.
+Math/cos a = (Math/sin (- (/ Math/PI 2.0) a))
```
Comment (developedby) [inline src/fun/builtins.bend:399]: A slightly faster implementation would be to implement this using the underlying hvm sine with -pi/2 as the second argument.

File: src/fun/builtins.bend [hunk 2]
```diff
@@ -383,3 +383,60 @@ Bytes/decode_ascii (List/Nil)       = (String/Nil)
...
+
+# Math/tan(a: f24) -> f24
+# Computes the tangent of the given angle in radians.
+hvm Math/tan:
+  ($([>>0x0] a) a)
+
+# Math/cot(a: f24) -> f24
+# Computes the cotangent of the given angle in radians.
+Math/cot a = (/ (Math/cos a) (Math/sin a))
```
Comment (developedby) [inline src/fun/builtins.bend:408]:
```suggestion
Math/cot a = (** (Math/tan a) -1.0)
```
to avoid duplicating `a`

Comment (developedby) [reply src/fun/builtins.bend:408]: or `(/ 1.0 (Math/tan a))`, don't know if either is better

File: src/fun/builtins.bend [hunk 3]
```diff
@@ -383,3 +383,60 @@ Bytes/decode_ascii (List/Nil)       = (String/Nil)
...
+  i_n = to_f24(to_i24(n))
+  if n == i_n:
+    return i_n
+  else:
+    return i_n + 1.0
+
+# Math/floor(n: f24) -> f24
+# Round float down to the nearest integer.
+def Math/floor(n):
+  i_n = to_f24(to_i24(n))
```
Comment (developedby) [inline src/fun/builtins.bend:438]: `to_i24` already rounds down, no?

**Review by developedby — CHANGES_REQUESTED** [review]
Can you add a round operation as well?

**Review by developedby — CHANGES_REQUESTED** [review]
Can you also add `atan`, `asin` and `acos`? You can calculate them from atan2

File: src/fun/builtins.bend (hunk 3 above)
Comment (imaqtkatt) [reply src/fun/builtins.bend:438]: Yes, but it does not work properly for negative values

Comment (developedby) [reply src/fun/builtins.bend:438]: Can you add some tests for negative values then? Also tests with already integer values (integers stored as floats, like 1.0, 2.0, etc)

**Review by developedby — APPROVED** [review]
